docs/reviewer-guide.md

```diff
@@ -41,6 +41,58 @@ Make a few passes over the code you want to review.
 * List ordering on CRD APIs matter. If two components need to update a list on the same object, make sure both components do it in a way that preserves the order of the list. For example, both virt-handler and virt-controller need to modify conditions on the VMI status. If both virt-handler and virt-controller are constantly changing the order of the conditions list, that will cause an update storm where both components are competing with one another to write changes.
 * Privileged node-level operations should be added to virt-handler and not virt-launcher to keep the privileges on virt-launcher at a minimum.
 
+## In-Depth PATCH/UPDATE considerations
+
+When trying to determine if an UPDATE or a PATCH should be used, the following has to be considered:
+
+### Updating `spec` or `metadata` sections with a controller
+
+Did the controller in question CREATE the object it operates on?
+- Yes: `Update` can be used
+- No: `Patch` with a narrow focus on the fields of interest needs to be used
+
+Reasoning:
+
+A controller which did not create an object can not have a full picture of the
+object which it operates on. Operating on a compatible API does not mean that
+the typed rest clients of k8s see all existing fields. Deserializing and
+serializing an object can therefore lead to lost information since a the
+final `Update` call would just remove the unknown fields.
```

Comment:
Seems pretty critical! Do you think it would feasible to write a test script that ensures nothing ever updates an object it did not create?

Reply:
I am not sure how this would look like. Other people and controllers are in general allowed to modify the spec. The critical point is just that if you did not crate the object, you may yourself use an older version of the API structs where you potentially remove as a side-effect fields on update. If your controller created the object, unless you downgrade, you know at least all the fields of the object which were used to create the object.